debezium-connector-oracle/src/main/java/io/debezium/connector/oracle/OracleValueConverters.java

```diff
@@ -194,6 +243,85 @@ protected Object convertNumeric(Column column, Field fieldDefn, Object data) {
         return super.convertNumeric(column, fieldDefn, data);
     }
 
+    protected Object convertNumericAsTinyInt(Column column, Field fieldDefn, Object data) {
+        if (data instanceof NUMBER) {
+            try {
+                data = ((NUMBER)data).byteValue();
+            }
+            catch (SQLException e) {
+                throw new RuntimeException("Couldn't convert value for column " + column.name(), e);
+            }
+        }
+
+        return convertTinyInt(column, fieldDefn, data);
+    }
+
+    protected Object convertNumericAsSmallInt(Column column, Field fieldDefn, Object data) {
+        if (data instanceof NUMBER) {
+            try {
+                data = ((NUMBER)data).shortValue();
+            }
+            catch (SQLException e) {
+                throw new RuntimeException("Couldn't convert value for column " + column.name(), e);
+            }
+        }
+
+        return super.convertSmallInt(column, fieldDefn, data);
+    }
+
+    protected Object convertNumericAsInteger(Column column, Field fieldDefn, Object data) {
+        if (data instanceof NUMBER) {
+            try {
+                data = ((NUMBER)data).intValue();
+            }
+            catch (SQLException e) {
+                throw new RuntimeException("Couldn't convert value for column " + column.name(), e);
+            }
+        }
+
+        return super.convertInteger(column, fieldDefn, data);
+    }
+
+    protected Object convertNumericAsBigInteger(Column column, Field fieldDefn, Object data) {
+        if (data instanceof NUMBER) {
+            try {
+                data = ((NUMBER)data).longValue();
+            }
+            catch (SQLException e) {
+                throw new RuntimeException("Couldn't convert value for column " + column.name(), e);
+            }
+        }
+
+        return super.convertBigInt(column, fieldDefn, data);
+    }
+
+    @Override
+    protected Object convertTinyInt(Column column, Field fieldDefn, Object data) {
+        if (data == null) {
+            data = fieldDefn.schema().defaultValue();
+        }
+        if (data == null) {
+            if (column.isOptional()) {
+                return null;
+            }
+            else {
+                return BYTE_FALSE;
```

Comment:
In this ocntext the constant should be called `BYTE_ZERO` otherwise it is a little bit misleading.

Reply:
Indeed; I had meant to remove it. Will do it.